### PR TITLE
Use Include What You Use

### LINKED ISSUE #240: Use Include What You Use
### Description

[Include What You Use](https://github.com/include-what-you-use/include-what-you-use) is a tool that checks if you have included all headers that you need but nothing more. Manually checking that gets more and more tedious with our growing project size, so having this tool would be very convenient. Also, there is native CMake support for it since CMake Version 3.3.

### PR DESCRIPTION
**Pull request by jeromehue**
### Description

Use [Include What You Use](https://github.com/include-what-you-use/include-what-you-use), via a new CMake preset. Additionally, a [mapping file](https://github.com/include-what-you-use/include-what-you-use/blob/master/docs/IWYUMappings.md) is provided.

Fixes #240

### CODE REVIEW

File: CMakePresets.json [hunk 1]
```diff
@@ -43,6 +43,13 @@
         "CMAKE_CXX_CLANG_TIDY": "clang-tidy;--header-filter=${sourceDir}/*"
       }
     },
+    {
+      "name": "include-what-you-use",
+      "hidden": true,
+      "cacheVariables": {
+        "CMAKE_CXX_INCLUDE_WHAT_YOU_USE": "include-what-you-use;-Xiwyu;--mapping_file=${sourceDir}/iwyu.imp;-Xiwyu;--no_fwd_decls"
```
Comment (PatrickKa) [inline CMakePresets.json:50]: Why is `--no_fwd_decls` needed? Aren't they preferable to including the whole header file?

File: README.md [hunk 2]
```diff
@@ -54,7 +54,8 @@ Docker image, the file should look something like the following:
           "dev-mode",
           "clang-tidy",
           "cppcheck",
-          "ci-unix"
+          "ci-unix",
+          "include-what-you-use"
```
Comment (PatrickKa) [inline README.md:58]: Why is it only added to the user presets but not the one used in CI?

Comment (jeromehue) [reply README.md:58]: Fixed by https://github.com/SpaceTeam/STS1_COBC_SW/pull/253/commits/f3240eef5d1841a70cb28167aa667d8965538130

File: CMakePresets.json (hunk 1 above)
Comment (PatrickKa) [reply CMakePresets.json:50]: We talked about it. It's fine.

**Pull request merged**